### PR TITLE
Communication phone number administration

### PR DESCRIPTION
**Pull request by pavelprystinka**
Phone administration is added to SDK

### CODE REVIEW

File: sdk/communication/Azure.Communication.Administration/README.md [hunk 1]
```diff
@@ -91,6 +93,126 @@ catch (RequestFailedException ex)
...
+
+### Phone plans overview
+
+Phone plans come in two types; Geographic and Toll-Free. Geographic phone plans are phone plans associated with a location, whose phone numbers' area codes are associated with the area code of a geographic location. Toll-Free phone plans are phone plans not associated location. For example, in the US, toll-free numbers can come with area codes such as 800 or 888.
+
+All geographic phone plans within the same country are grouped into a phone plan group with a Geographic phone number type. All Toll-Free phone plans within the same country are grouped into a phone plan group.
+
+### Searching and Acquiring numbers
+
+Phone numbers search can be search through the search creation API by providing a phone plan id, an area code and quantity of phone numbers. The provided quantity of phone numbers will be reserved for ten minutes. This search of phone numbers can either be cancelled or purchased. If the search is cancelled, then the phone numbers will become available to others. If the search is purchased, then the phone numbers are acquired for the Azure resources.
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Administration/README.md:106]: a lot of "search" :D "Phone numbers search can be search through the search creation API"

File: sdk/communication/Azure.Communication.Administration/README.md [hunk 2]
```diff
@@ -91,6 +93,126 @@ catch (RequestFailedException ex)
...
 
+
+
+### Phone plans overview
+
+Phone plans come in two types; Geographic and Toll-Free. Geographic phone plans are phone plans associated with a location, whose phone numbers' area codes are associated with the area code of a geographic location. Toll-Free phone plans are phone plans not associated location. For example, in the US, toll-free numbers can come with area codes such as 800 or 888.
+
+All geographic phone plans within the same country are grouped into a phone plan group with a Geographic phone number type. All Toll-Free phone plans within the same country are grouped into a phone plan group.
+
+### Searching and Acquiring numbers
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Administration/README.md:104]: "a"cquiring

File: sdk/communication/Azure.Communication.Administration/README.md [hunk 3]
```diff
@@ -91,6 +93,126 @@ catch (RequestFailedException ex)
...
+
+Phone numbers search can be search through the search creation API by providing a phone plan id, an area code and quantity of phone numbers. The provided quantity of phone numbers will be reserved for ten minutes. This search of phone numbers can either be cancelled or purchased. If the search is cancelled, then the phone numbers will become available to others. If the search is purchased, then the phone numbers are acquired for the Azure resources.
+
+### Configuring / Assigning numbers
+
+Phone numbers can be assigned to a callback URL via the configure number API. As part of the configuration, you will need an acquired phone number, callback URL and application id.
+
+## Examples
+
+### Get Countries
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Administration/README.md:114]: Get list of the countries that are supported by the service

File: sdk/communication/Azure.Communication.Administration/README.md [hunk 4]
```diff
@@ -91,6 +93,126 @@ catch (RequestFailedException ex)
...
+PhoneNumberAdministrationClient client = new PhoneNumberAdministrationClient(connectionString);
+Pageable<PhoneNumberCountry> countries = client.GetAllSupportedCountries();
+
+foreach (var country in countries)
+{
+    Console.WriteLine($"Country code {country.CountryCode}, Country name: {country.LocalizedName}");
+}
+```
+
+### Get Phone Plan Groups
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Administration/README.md:127]: Capitalization (Same for other sections): Get phone plan groups

File: sdk/communication/Azure.Communication.Administration/src/Azure.Communication.Administration.csproj [hunk 5]
```diff
@@ -7,7 +7,7 @@
     <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
     <EnableApiCompat>false</EnableApiCompat>
     <!-- These supressions should be removed in a production library -->
-    <NoWarn>$(NoWarn);AZC0001</NoWarn>
+    <NoWarn>$(NoWarn);AZC0001;AZC0012</NoWarn>
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Administration/src/Azure.Communication.Administration.csproj:10]: Why do we need to disable AZC0012?

File: sdk/communication/Azure.Communication.Administration/src/PhoneNumberSearchCancelOperation.cs [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Communication.Administration.Models;
+
+namespace Azure.Communication.Administration
+{
+    /// <summary>
+    /// Represent a long-running phone number search cancellation operation.
+    /// </summary>
+    public class PhoneNumberSearchCancelOperation : Operation<PhoneNumberSearch>
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Administration/src/PhoneNumberSearchCancelOperation.cs:16]: I thought we decided to remove all the operations and bring them back after the redesign

**Review by RezaJooyandeh — COMMENTED** [review]
Also let's add to the changelog that PhoneNumberAdministrationClient is added to this version and what it does

File: sdk/communication/Azure.Communication.Administration/src/Azure.Communication.Administration.csproj (hunk 5 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/src/Azure.Communication.Administration.csproj:10]: Otherwise we can not have class Capability

Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.Administration/src/Azure.Communication.Administration.csproj:10]: Let's use `CodeGenModel` attribute and rename it to `PhoneNumberCapability"

File: sdk/communication/Azure.Communication.Administration/README.md (hunk 1 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/README.md:106]: I will leave one less "search":
Phone numbers search can be performed through the search creation API by providing a phone plan id

File: sdk/communication/Azure.Communication.Administration/CHANGELOG.md [hunk 7]
```diff
@@ -1,7 +1,7 @@
 # Release History
 
 ## 1.0.0-beta.2 (Unreleased)
-
+Added phone number administration. For more information, please see the [README][read_me] and [documentation][documentation].
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Administration/CHANGELOG.md:4]: I don't think we have to add the "For more information, ..." part. That's already in the previous version changelog, and the README itself references the docs.

Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.Administration/CHANGELOG.md:4]: I think for C# is different.
https://www.nuget.org/packages/Azure.Communication.Administration/
It seems that it picks things up from the csproj itself. But let's do that afterwards

File: sdk/communication/Azure.Communication.Administration/src/Azure.Communication.Administration.csproj (hunk 5 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/src/Azure.Communication.Administration.csproj:10]: Generated code makes class Capability that is eventually causing this error
If I add CodeGenModel and decorate with attribute that will export it as PhoneNumberCapability.
BTW this class used to have name PhoneNumberCapability on the server side and in SDK we used to use CodeGenModel to rename it to Capability.

**Comment by RezaJooyandeh** [comment]
@fearthecowboy and @pakrym based on the [build failure](https://dev.azure.com/azure-sdk/public/_build/results?buildId=562426&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=bdeefc16-b669-5ebd-ad94-a2c19ade53b0), it seems that autorest is merging swaggers together and [assumes they are the same clients](https://github.com/Azure/autorest/blob/af09368e6010e3f166bbafc6b1ccbc2562388864/core/lib/pipeline/plugins/composer.ts#L57). Does it mean that we do not support two different clients in one assembly anymore?

**Comment by pakrym** [comment]
We do, but not via having multiple titles. Clients are created based on the operation id prefix (`"operationId": "PhoneNumberAdministration_GetAllPhoneNumbers"` would result in `PhoneNumberAdministration` client).

You can suppress the title issue by defining a title in `autorest.md`, the title is only used for the "default" client that consists of operations that didn't have a prefix.

**Pull request merged**